pandas/core/dtypes/common.py

```diff
@@ -1328,7 +1328,8 @@ def is_bool_dtype(arr_or_dtype) -> bool:
         # we don't have a boolean Index class
         # so its object, we need to infer to
         # guess this
```

Comment:
this can be removed now I think (or at least consolidated with the new comment you added below)